### PR TITLE
[Refactor][Manifest] flip elementwise multi-input/clamp/masked_fill ops to status: implemented (follow-up to #1121)

### LINKED ISSUE #1128: [Refactor][Manifest] flip 7 elementwise multi-input/clamp/masked_fill ops to status: implemented (follow-up to #1121)
### Symptom / Motivation

Discovered during PR #1121 ([Feat][Ops] align where/clamp/masked_fill multi-input ops with PyTorch API). The PR aligned op classes and TileLang kernels with the manifest specs but, per `.claude/rules/manifest-trust-model.md`, did not touch the manifest. The seven affected ops still carry `status: spec-only` even though their code now fully conforms.

Concretely, with the merge of #1121:

- `tileops/manifest/elementwise_multi_input.yaml` — `WhereFwdOp` (status: spec-only)
- `tileops/manifest/elementwise_unary_activation.yaml` — `ClampFwdOp`, `ClampScalarFwdOp`, `ClampMinFwdOp`, `ClampMaxFwdOp` (all status: spec-only)
- `tileops/manifest/elementwise_binary.yaml` — `MaskedFillFwdOp`, `MaskedFillScalarFwdOp` (status: spec-only)

The stale status hides conformance from anything that filters on `status` (validators, audit reports, scaffolders) and prevents downstream tooling from treating these ops as production-ready.

### Root Cause Analysis

Manifest-trust-model splits implementation and manifest changes across separate PRs with separate human review. The flip was deliberately deferred from #1121 to keep the trust boundary clean.

### Related Files

- `tileops/manifest/elementwise_multi_input.yaml`
- `tileops/manifest/elementwise_unary_activation.yaml`
- `tileops/manifest/elementwise_binary.yaml`
- `scripts/validate_manifest.py`

## Goal

Flip `status: spec-only` → `status: implemented` for the seven ops aligned by PR #1121, in a manifest-only PR.

## Plan

1. For each of the seven ops above, edit only the `status:` line in its manifest entry. Use `ruamel.yaml` to preserve comments and key order; do not touch any other field.
2. Drop or shorten the now-stale `# status: spec-only — current ... does not conform ...` comment block above the status line where it exists, since the divergence it describes is closed.
3. Run `python scripts/validate_manifest.py` and confirm zero new errors compared to `upstream/main`.
4. Open a manifest-only PR titled along the lines of `[Refactor][Manifest] flip elementwise multi-input/clamp/masked_fill ops to status: implemented (follow-up to #1121)`. Body links back to #1121 for the conformance evidence.

## Constraints

- Must not regress PR #1121.
- Manifest-only diff — do not edit any code under `tileops/ops/`, `tileops/kernels/`, `tests/`, or `benchmarks/`.
- Op signatures, shape rules, dtype combos, and roofline fields are out of scope; only the `status` field (and its now-stale rationale comment) flips.

## Acceptance Criteria

- [ ] AC-1: All seven ops list `status: implemented` in their manifest files.
- [ ] AC-2: `python scripts/validate_manifest.py` passes.
- [ ] AC-3: No diff outside `tileops/manifest/`.
- [ ] AC-4: PR description cites #1121 as the conformance evidence.

### PR DESCRIPTION
**Pull request by lcy-seso**
Closes #1128

## Summary

- Flip `status: spec-only` → `status: implemented` for the seven multi-input / clamp / masked_fill elementwise ops aligned by PR #1121: `WhereFwdOp`, `ClampFwdOp`, `ClampScalarFwdOp`, `ClampMinFwdOp`, `ClampMaxFwdOp`, `MaskedFillFwdOp`, `MaskedFillScalarFwdOp`.
- Drop the now-stale `# status: spec-only — current ... does not conform ...` rationale comments that PR #1121 made obsolete.
- Manifest-only follow-up per [`.claude/rules/manifest-trust-model.md`](.claude/rules/manifest-trust-model.md): manifest changes ship in their own PR, separate from the implementation PR (#1121) that brought the code into conformance.

## Conformance evidence

PR #1121 (`[Feat][Ops] align where/clamp/masked_fill multi-input ops with PyTorch API`, merged) is the conformance evidence for AC-4 of issue #1128. It aligned the op signatures, shape rules, and dtype handling of these seven ops with the manifest spec, so flipping `status` from `spec-only` to `implemented` here is purely a metadata follow-up.

## Test plan

- [x] AC-1: All seven ops (`WhereFwdOp`, `ClampFwdOp`, `ClampScalarFwdOp`, `ClampMinFwdOp`, `ClampMaxFwdOp`, `MaskedFillFwdOp`, `MaskedFillScalarFwdOp`) list `status: implemented` in their manifest files.
- [x] AC-2: `python scripts/validate_manifest.py` passes.
- [x] AC-3: No diff outside `tileops/manifest/`.
- [x] AC-4: PR description cites #1121 as the conformance evidence.

### Validator: baseline vs HEAD

Both `upstream/main` baseline and this branch HEAD report **0 errors, exit 0, "All manifest checks passed."** No new validator errors introduced.

The post-flip warnings (`kernel_map missing`, parity checks skipped, `bench_manifest_driven`) are non-blocking and match the pattern already accepted for other `status: implemented` ops such as `RMSNormFwdOp` and `SoftmaxFwdOp`.

## Regression

Manifest-only diff (3 files, +7/-44). No code under `tileops/ops/`, `tileops/kernels/`, `tests/`, or `benchmarks/` is touched, so there is no runtime surface to regress. PR #1121 remains the source of behavioral coverage for these ops.

## Additional context

- Issue: #1128
- Implementation PR (conformance evidence): #1121
- Trust model: [`.claude/rules/manifest-trust-model.md`](.claude/rules/manifest-trust-model.md) — manifest changes are reviewed separately from implementation changes.

### CODE REVIEW

File: tileops/manifest/elementwise_binary.yaml [hunk 1]
```diff
@@ -60,12 +60,7 @@ MaskedFillFwdOp:
   # value form lands as MaskedFillScalarFwdOp below per the
   # "No Optional[Tensor]" manifest rule for variant splits.
   family: elementwise
-  # status: spec-only — no kernel currently implements the 0-dim Tensor
-  # value form; tileops/ops/elementwise.py:MaskedFillFwdOp implements
-  # the scalar form (see MaskedFillScalarFwdOp). Code conforms to this
-  # spec in a follow-up PR per the manifest trust model
-  # (.claude/rules/manifest-trust-model.md).
-  status: spec-only
+  status: implemented
```
Comment (Ibuki-wind) [inline tileops/manifest/elementwise_binary.yaml:63]: `MaskedFillFwdOp` and `MaskedFillScalarFwdOp` now claim `status: implemented`, but their `source` blocks still lack `source.kernel_map`; implemented entries must declare the Op-to-Kernel dispatch map so the manifest remains authoritative → add `source.kernel_map` entries for both flipped masked-fill entries, matching each class default kernel map.

File: tileops/manifest/elementwise_multi_input.yaml [hunk 2]
```diff
@@ -8,23 +8,12 @@
...
-  # Code conforms to this spec in a follow-up PR per the manifest trust
-  # model (.claude/rules/manifest-trust-model.md).
-  #
   # TODO(spec-followup): PyTorch's ``torch.where`` also performs implicit
   # type promotion between ``input`` and ``other``. The manifest currently
   # constrains them to ``same_as(input)``; once the kernel grows an input
   # cast path, relax ``other`` to the full float union and document the
   # promotion contract here.
-  status: spec-only
+  status: implemented
```
Comment (Ibuki-wind) [inline tileops/manifest/elementwise_multi_input.yaml:16]: `WhereFwdOp` now claims `status: implemented`, but its `source` block still lacks `source.kernel_map`; implemented entries must declare the Op-to-Kernel dispatch map so the manifest remains authoritative → add a `source.kernel_map` entry for `WhereFwdOp` matching the class default kernel map.

File: tileops/manifest/elementwise_unary_activation.yaml [hunk 3]
```diff
@@ -400,12 +400,7 @@ ClampFwdOp:
   # single-bound min-only form, and the single-bound max-only form each
   # land as their own variant_of entries below.
   family: elementwise
-  # status: spec-only — current tileops/ops/elementwise.py:ClampFwdOp
-  # implements the *scalar* form (see ClampScalarFwdOp below) and does
-  # not conform to this Tensor-bound primary spec. Code conforms to this
-  # spec in a follow-up PR per the manifest trust model
-  # (.claude/rules/manifest-trust-model.md).
-  status: spec-only
+  status: implemented
```
Comment (Ibuki-wind) [inline tileops/manifest/elementwise_unary_activation.yaml:403]: `ClampFwdOp`, `ClampScalarFwdOp`, `ClampMinFwdOp`, and `ClampMaxFwdOp` now claim `status: implemented`, but their `source` blocks still lack `source.kernel_map`; implemented entries must declare the Op-to-Kernel dispatch map so the manifest remains authoritative → add `source.kernel_map` entries for every flipped clamp entry, matching each class default kernel map.

**Review by Ibuki-wind — CHANGES_REQUESTED** [review]
### Overall
Implemented status flips are missing required manifest dispatch metadata before the claim is trustworthy.

File: tileops/manifest/elementwise_binary.yaml (hunk 1 above)
Comment (lcy-seso) [reply tileops/manifest/elementwise_binary.yaml:63]: Adopted. Added source.kernel_map with masked_fill_tensor_value: MaskedFillTensorValueFwdKernel for MaskedFillFwdOp and masked_fill: MaskedFillFwdKernel for MaskedFillScalarFwdOp. See e8dea07.

File: tileops/manifest/elementwise_multi_input.yaml (hunk 2 above)
Comment (lcy-seso) [reply tileops/manifest/elementwise_multi_input.yaml:16]: Adopted. Added source.kernel_map with where: WhereFwdKernel. See e8dea07.

File: tileops/manifest/elementwise_unary_activation.yaml (hunk 3 above)
Comment (lcy-seso) [reply tileops/manifest/elementwise_unary_activation.yaml:403]: Adopted. Added source.kernel_map for all four clamp entries — clamp_tensor: ClampTensorFwdKernel for ClampFwdOp, ClampMinFwdOp, ClampMaxFwdOp; clamp: ClampFwdKernel for ClampScalarFwdOp. See e8dea07.

**Review by Ibuki-wind — APPROVED** [review]
Clean — no issues.
